doc/source/mergingdata.rst

```diff
@@ -117,18 +117,186 @@ Note more complicated spatial relationships can be studied by combining geometri
 Sjoin Performance
 ~~~~~~~~~~~~~~~~~~
 
-Existing spatial indexes on either `left_df` or `right_df` will be reused when performing an ``sjoin``. If neither df has a spatial index, a spatial index will be generated for the longer df. If both have a spatial index, the `right_df`'s index will be used preferentially. Performance of multiple sjoins in a row involving a common GeoDataFrame may be improved by pre-generating the spatial index of the common GeoDataFrame prior to performing sjoins using ``df1.sindex``.
+Most of the computation is done within the spatial index queries. The spatial index is always queried as follows:
 
 .. code-block:: python
 
-    df1 = # a GeoDataFrame with data
-    df2 = # a second GeoDataFrame
-    df3 = # a third GeoDataFrame
+   right_df.sindex.query_bulk(left_df.geometry, op)
 
-    # pre-generate sindex on df1 if it doesn't already exist
-    df1.sindex
+This translates to:
+
+.. code-block:: python
+
+   geom_in_left_df.op(geom_in_right_df)
+
+For ``sjoin``, there are two concrete choices which can be made:
+
+**Order of left_df and right_df**
+
+For predicates where the order does not matter (ex: ``intersects``) simply flipping the GeoDataFrames and the ``how`` parameter can yield great performance improvements.
+
+.. code-block:: python
+
+   # changing
+   sjoin(left_df=df1, right_df=df2, how="left", op='intersects')
+   # to
+   sjoin(left_df=df2, right_df=df1, how="right", op='intersects')
+   # yields the same result but possibly different performance
+
+There are many factors that may influence the performance of the query, some of which are:
+* Complexity of geometries. Some operations may handle complex geometries better than others. For example, ``polygon.intersects(point)`` will perform differently than ``point.intersects(polygon)``. Swapping ``left_df`` and ``right_df`` will change which way the comparison is made.
```

Comment:
The `*` formatting of the list is broken if you try to build the page with sphinx. You need a space above.
```suggestion

* Complexity of geometries. Some operations may handle complex geometries better than others. For example, ``polygon.intersects(point)`` will perform differently than ``point.intersects(polygon)``. Swapping ``left_df`` and ``right_df`` will change which way the comparison is made.
```

Reply:
Fixed